debezium-core/src/main/java/io/debezium/transforms/ExtractNewRecordStateConfigDefinition.java

```diff
@@ -67,6 +67,71 @@ public static DeleteHandling parse(String value, String defaultValue) {
         }
     }
 
+    public enum DeleteTombstoneHandling implements EnumeratedValue {
+        DROP("drop"),
+        TOMBSTONE("tombstone"),
+        REWRITE("rewrite"),
+        REWRITE_WITH_TOMBSTONE("rewrite-with-tombstone");
+
+        private final String value;
+
+        DeleteTombstoneHandling(String value) {
+            this.value = value;
+        }
+
+        @Override
+        public String getValue() {
+            return value;
+        }
+
+        /**
+         * Determine if the supplied value is one of the predefined options.
+         *
+         * @param value the configuration property value; may not be null
+         * @return the matching option, or null if no match is found
+         */
+        public static DeleteTombstoneHandling parse(String value) {
+            if (value == null) {
+                return null;
+            }
+            value = value.trim();
+            for (DeleteTombstoneHandling option : DeleteTombstoneHandling.values()) {
+                if (option.getValue().equalsIgnoreCase(value)) {
+                    return option;
+                }
+            }
+            return null;
+        }
+
+        /**
+         * Determine if the supplied value is one of the predefined options.
+         *
+         * @param value the configuration property value; may not be null
+         * @param defaultValue the default value; may be null
+         * @return the matching option, or null if no match is found and the non-null default is invalid
+         */
+        public static DeleteTombstoneHandling parse(String value, String defaultValue) {
+            DeleteTombstoneHandling mode = parse(value);
+            if (mode == null && defaultValue != null) {
+                mode = parse(defaultValue);
+            }
+            return mode;
+        }
+    }
+
+    public static final Field HANDLE_TOMBSTONE_DELETES = Field.create("delete.tombstone.handling.mode")
+            .withDisplayName("Handle delete records")
+            .withEnum(DeleteTombstoneHandling.class)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.MEDIUM)
+            .optional()
+            .withDescription("How to handle delete records. Options are: "
+                    + "drop - will remove both delete and tombstone,"
+                    + "tombstone - will convert delete to tombstone and remove tombstone (the default),"
+                    + "rewrite - will convert delete and remove tombstone, __deleted field is added to records,"
```

Comment:
```suggestion
                    + "rewrite - add a `__deleted"` field with the value `true` to the record and remove tombstone and delete event from the stream,"
```
`ExtractNewRecordState` also makes sense to use on the sink side as well, right? Then this option should be marked as only use-able on the source side, no? Or can it be used on the sink side as well? I don't think it works there in the same way as on the sink side, does it? It can't combine multiple messages into one there, right?


When I was writing this I was also confused by the difference between the physical thing happening - the delete operation of a record in the database and the name of the message that Debezium produces to reflect that delete operation in the stream, which we also call a `DELETE event`. I would call the operation happening in the database the `delete event`. But we use `delete event` here as a name for  the delete message that Debezium emits to reflect that event. That makes it hard for me to understand and precisely identify when we mean what.
@Naros  @jpechane @roldanbob wdyt?
Should we maybe change the wording and call the operation `delete event` and the Debezium delete record as `delete message` or something? Maybe it is just me but I find this easy to confuse. :man_facepalming:

Reply:
In my way of thinking, we need to use language to precisely differentiate between events and event records. 
That is, the original operation that is applied to a table is an event (`delete` event). 
When Debezium captures such an event from a table, it then emits a corresponding event record (i.e., a `delete event record`).  
I believe this matches up with the terminology in most of the documentation.